string node_name
string name
---